.chloggen/mx-psi_type-mismatch.yaml

```diff
@@ -0,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
```

Comment:
Given the breakage on contrib I am not sure if this should be marked as a 'breaking' API change. I don't think it is (I feel like it's a misuse of the API to not pass a matching `component.Type`), and the breakage should only happen on tests but still it may be a bit confusing

Reply:
I would call it a bug_fix in that case

Reply:
Alright, fixed!